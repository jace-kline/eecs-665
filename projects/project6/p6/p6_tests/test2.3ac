[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN v LOCALS]
a (formal)
ptr (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
tmp3 (tmp)
tmp4 (tmp)
tmp5 (tmp)
[END v LOCALS]
lbl_fun_v: enter v
getarg 1 [a]
[tmp0] := a
[ptr] := [tmp0]
tmp1 := [ptr]
[tmp1] := 12
[tmp2] := 4 MULT64 8
[tmp3] := [ptr] ADD64 [tmp2]
tmp4 := [tmp3]
[tmp4] := 16
tmp5 := [ptr]
setret [tmp5]
goto lbl_0
lbl_0: leave v

