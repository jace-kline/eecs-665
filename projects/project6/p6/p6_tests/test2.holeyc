int v(int a) {
    intptr ptr;
    ptr = ^a;
    @ptr = 12;
    ptr[4] = 16;
    return @ptr;
}