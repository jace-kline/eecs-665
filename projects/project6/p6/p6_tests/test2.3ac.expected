[BEGIN GLOBALS]
[END GLOBALS]
[BEGIN v LOCALS]
a (formal)
ptr (local)
tmp0 (tmp)
tmp1 (tmp)
tmp2 (tmp)
tmp3 (tmp)
tmp4 (tmp)
[END v LOCALS]
lbl_fun_v:  enter v
            getarg 1 [a]
            [tmp0] := a
            [ptr] := [tmp0]
            tmp1 := [ptr]
            [tmp1] := 12
            [tmp2] := [ptr] ADD64 4
            tmp3 := [tmp2]
            [tmp3] := 16
            tmp4 := [ptr]
            setret [tmp4]
            goto lbl_0
lbl_0:      leave v
